sherpa/astro/ui/utils.py

```diff
@@ -13558,8 +13577,11 @@ def sample_flux(self, modelcomponent=None, lo=None, hi=None, id=None,
             samples = self.calc_energy_flux(lo=lo, hi=hi, id=id,
                                             bkg_id=bkg_id)
         else:
+            # NOTE: the samples are drawn from the full model expression
+            #       as this is how it was originally written
+            #
             samples = self.sample_energy_flux(lo=lo, hi=hi, id=id, num=niter,
-                                              scales=scales, clip='hard',
+                                              scales=scales, clip='soft',
```

Comment:
@anetasie left a general comment asking why you use soft here. My understanding is that soft limits set the values that are allowed for the fitting, e.g. I have prior knowledge that the temperature of the star is > 0.3 keV (let's say, I have a UV spectrum), so I set the soft limit or 0.3 keV. However, that should not stop the error bar from going below 0.3 keV, so `sample_energy_flux` should be allowed to go down to the hard limits (obviously, it can't go beyond the hard limits, because I can't evaluate the model there). Or maybe my interpretation of what "soft" limits in Sherpa are is wrong?

Reply:
Yes, this was my basic issue - how to use soft/hard limits in `sample_flux`. Is the intended answer use `sample_energy_flux`?